src/utils/popups.js

```diff
@@ -55,15 +57,52 @@ const wcLinkPopupHTML = (title, displayProps) => {
           ${downstreamWCLinks}`;
 };
 
+const fitMapToFeatures = (map, features) => {
+  var bounds = new mapboxgl.LngLatBounds();
+
+  features.forEach((feature) => {
+    if (feature.geometry.type === "LineString") {
+      feature.geometry.coordinates.forEach((coords) => {
+        bounds.extend(coords);
+      });
+    } else if (feature.geometry.type === "MultiLineString") {
+      feature.geometry.coordinates.forEach((line) => {
+        line.forEach((coords) => {
+          bounds.extend(coords);
+        });
+      });
+    }
+  });
+
+  map.fitBounds(bounds, { padding: 20 });
+};
+
 const getUpstream = async (id) => {
   const url = `${waterNetworkAPIBase}/collections/WatercourseLink/items/${id}/upstream`;
   return await getURL(url);
 };
 // TODO: catch 404s
 const highlightUpstreamWatercourseLinks = async (id, map) => {
-  await getUpstream(id).then((watercourseLinks) => {
+  await getUpstream(id).then(async (upstreamWatercourseLinks) => {
     closePopup();
-    map.getSource("highlightWatercourseLink").setData(watercourseLinks);
+    const renderedWcLinks = map.querySourceFeatures("watercourseLinks");
+    const renderedHNs = map.querySourceFeatures("hydroNodes");
+
+    const allWcLinks = mergeFeatures(upstreamWatercourseLinks, renderedWcLinks);
+
+    map.getSource("watercourseLinks").setData(allWcLinks);
+    map.getSource("hydroNodes").setData({
+      type: "FeatureCollection",
+      features: [...renderedHNs],
+    });
```

Comment:
Doing a similar thing here so that we don't end up with spooky ghost/floating hydro nodes 👻 

example of lonely hydro nodes:
![image](https://user-images.githubusercontent.com/32230328/197269756-ade4567f-483d-4c46-b8fa-d7c2084c5dc5.png)
![image](https://user-images.githubusercontent.com/32230328/197269768-242f3d2f-6409-4f34-bca7-4dea68ee274f.png)
![image](https://user-images.githubusercontent.com/32230328/197269776-3589aaa4-c9d5-499d-acf9-5702369ce32d.png)